cargo-dist/src/config/version.rs

```diff
@@ -0,0 +1,67 @@
+//! For determining which configuration version we're using.
+
+use axoasset::SourceFile;
+use camino::Utf8PathBuf;
+use serde::Deserialize;
+
+use crate::{DistError, DistResult};
+
+const MAX_VERSION: i64 = 1;
+
+// Extremely minimal struct designed to differentiate between config versions.
+// V0 does not have the `config-version` field, so will fail to parse.
+// V1+ should have it, so will parse, and contain a `config_version` field.
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct FauxDistTable {
+    #[allow(dead_code)]
+    config_version: i64,
+}
+
+#[derive(Deserialize)]
+struct FauxConfig {
+    #[allow(dead_code)]
+    dist: FauxDistTable,
+}
+
+/// Return the default versioni.
+pub fn default() -> i64 {
+    1
+}
+
+/// Return the config version used for the root workspace.
+pub fn get_version() -> DistResult<i64> {
+    let workspaces = super::get_project()?;
+    let root_workspace = workspaces.root_workspace();
+
+    get_version_for_manifest(root_workspace.manifest_path.to_owned())
+}
+
+/// Given a path to a dist manifest (e.g. `dist-workspace.toml`), returns
+/// the config version being used.
+pub fn get_version_for_manifest(dist_manifest_path: Utf8PathBuf) -> DistResult<i64> {
+    if dist_manifest_path.file_name() != Some("dist-workspace.toml") {
+        // If the manifest is in Cargo.toml or dist.toml, we're
```

Comment:
this may not be strictly true because there was a brief period of time when folks put all config in a dist.toml (including workspace). i think practically tho, this is true, but wanted to call out the formal exception.